projects/igniteui-angular/src/lib/combo/combo.component.spec.ts

```diff
@@ -2338,1188 +1031,1591 @@ describe('igxCombo', () => {
+            fixture.detectChanges();
             tick();
-            expect(combo.data.length).toEqual(initialDataLength + 3);
-            expect(combo.dropdown.items.length).toEqual(4); // Add Item button is included
-            expect(combo.dropdown.headers.length).toEqual(1);
-            expect(combo.dropdown.headers[0].element.nativeElement.innerText).toEqual(fallBackGroup);
+            const data = fixture.componentInstance.items;
+            expect(combo.selectedItems()).toEqual(component.selectedItems);
+            combo.selectItems([...data].splice(1, 3), true);
+            fixture.detectChanges();
+            expect(fixture.componentInstance.selectedItems).toEqual([...data].splice(1, 3));
         }));
     });
+    describe('Dropdown tests: ', () => {
+        describe('complex data dropdown: ', () => {
+            let dropdown: IgxComboDropDownComponent;
+            configureTestSuite();
+            beforeAll(async(() => {
+                TestBed.configureTestingModule({
+                    declarations: [
+                        IgxComboSampleComponent
+                    ],
+                    imports: [
+                        IgxComboModule,
+                        NoopAnimationsModule,
+                        IgxToggleModule,
+                        ReactiveFormsModule,
+                        FormsModule
+                    ]
+                }).compileComponents();
+            }));
+            beforeEach(fakeAsync(() => {
+                fixture = TestBed.createComponent(IgxComboSampleComponent);
+                fixture.detectChanges();
+                combo = fixture.componentInstance.combo;
+                dropdown = combo.dropdown;
+                input = fixture.debugElement.query(By.css(`.${CSS_CLASS_INPUTGROUP}`));
+            }));
+            it('should properly call dropdown navigatePrev method', fakeAsync(() => {
+                expect(dropdown.focusedItem).toBeFalsy();
+                expect(dropdown.focusedItem).toEqual(null);
+                expect(combo.collapsed).toBeTruthy();
+                combo.toggle();
+                tick();
+                fixture.detectChanges();
+                expect(document.activeElement).toEqual(combo.searchInput.nativeElement);
+                expect(combo.collapsed).toBeFalsy();
+                combo.handleKeyUp(new KeyboardEvent('keyup', UIInteractions.arrowDownEvent));
+                fixture.detectChanges();
+                expect(dropdown.focusedItem).toBeTruthy();
+                expect(dropdown.focusedItem.itemIndex).toEqual(0);
+                expect((combo as any).virtDir.state.startIndex).toEqual(0);
+                dropdown.navigatePrev();
+                tick();
+                fixture.detectChanges();
+                expect(document.activeElement).toEqual(combo.searchInput.nativeElement);
+                combo.handleKeyUp(new KeyboardEvent('keyup', UIInteractions.arrowDownEvent));
```

Comment:
In the UIInteractions you can create a method getKeyUpEvent, similar to the triggerEventHandlerKeyDown